.github/workflows/coverage_report.yml

```diff
@@ -0,0 +1,62 @@
+name: Add coverage comment on PR
+
+on:
+  workflow_run:
+    workflows: ["Test and lint"]
+    types:
+      - completed
+
+permissions: {}
+
+jobs:
+  cov-report:
+    permissions:
+        contents: read
+        pull-requests: write
+    runs-on: ubuntu-latest
+    if: >
+      github.event.workflow_run.event == 'pull_request'
+      && github.repository == github.event.workflow_run.repository.full_name
```

Comment:
Unfortunately I don't think this works -- `workflow_run` always runs in the context of the "parent" repository, so the second hand of this `&&` expression always evaluate to `true` (I think).

Reply:
(But also, if the intention is to limit this to first-party PRs, then I think this `workflow_run` can be removed entirely and you can move these steps with their first-party check to `ci.yml` instead 🙂)

Reply:
For what it's worth, if this works only for first-party PRs that would be a dealbreaker to me; we get lots of third-party PRs and even maintainer PRs often come from forks.